.github/workflows/sam-pull-request.yml

```diff
@@ -16,28 +16,39 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Required for SonarQube scanning
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'corretto'
+          cache: gradle
 
-# Uncomment when we have a SONAR_TOKEN
-#      - name: Run SonarCloud Scan
-#        uses: SonarSource/sonarcloud-github-action@master
-#        env:
-#          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-#          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-#
-#      - name: SonarQube Quality Gate check
-#        id: sonarqube-quality-gate-check
-#        uses: sonarsource/sonarqube-quality-gate-action@master
-#        timeout-minutes: 5
-#        env:
-#          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-#
-#      - name: SonarQube Quality Gate Status value
-#        run: echo "The Quality Gate status is ${{ steps.sonarqube-quality-gate-check.outputs.quality-gate-status }}"
-
-      - name: Set up Python 3.8
-        uses: actions/setup-python@v4
+      - name: Cache Gradle packages
+        uses: actions/cache@v3
         with:
-          python-version: 3.8
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: ${{ runner.os }}-gradle
+
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v3
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: Build and unit tests
+        run: ./gradlew clean build
+
+      - name: Perform Static Analysis
+        if: ${{ github.actor != 'dependabot[bot]' }}
```

Comment:
So we don't do this on dependabot? This feels...odd, but presumably there's a reason?

Reply:
Nope this was just while I tested, removed